### ISSUE TITLE
Argument metavar parameter not used with exception message

### ISSUE THREAD

**Issue by cdunklau** [pallets/click#598]
#273 exists (again/still) in Click 6.6:
# Code

``` python
import click
print click.__version__

@click.command()
@click.argument('to_say', metavar='message')
def foo(to_say):
    print to_say


if __name__ == '__main__':
    foo()
```
# Expected Output

```
6.6
Usage: foo.py [OPTIONS] message

Error: Missing argument "message".
```
# Observed Output

```
6.6
Usage: foo.py [OPTIONS] message

Error: Missing argument "to_say".
```

**Comment by cdunklau** [comment]
Still an issue in 6.7

**Comment by haizaar** [comment]
Indeed an issue - should've been `type` instead of `backup_type`

```python
$ python /tmp/bug.py
Usage: bug.py [OPTIONS] type

Error: Missing argument "backup_type".

$ cat /tmp/bug.py
import click

@click.command('mssql')
@click.argument('backup_type', metavar='type')
def mssql(backup_type):
        click.echo('Running the backup...')
mssql()

$ pip show click
Name: click
Version: 6.7

$ python --version
Python 3.6.4
```

**Comment by LokiHokie** [comment]
Fixed #709
